### PR TITLE
Read-only mode for controllers

### LINKED ISSUE #6181: Forbid AST/execution context changes in controllers when read-only flag is set
All changes from View must result in an error if `read-only` mode is enabled. The error should be printed in logs (until better notifications are ready). It is necessary to prevent accidental bugs when the view handles the modification despite `read-only` mode.

### Acceptance criteria

```
GIVEN an open IDE with `read-only` mode active (can be controlled by a temporary shortcut or feature-flag)
WHEN I edit any node in graph editor or in code editor
THEN the error is printed in the log
AND The node restores its content

WHEN I create a new node using Tab key or Plus button
THEN the error is printed in the log
AND No node is being created

WHEN I use CB to preview suggestions
THEN The error is printed in the log
AND The suggestion is not applied (I can't see code modifications in the code editor)

WHEN I rename the project
THEN The error is printed in the log
AND The name stays the same
```

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

Closes #6181 

- Added a read-only flag for the project model (now controlled by a temporary shortcut).
- Renaming the project, editing the code, connecting and disconnecting nodes, and collapsing, and navigating through collapsed nodes are forbidden and will result in the error message in the console.
- Some of the above actions produce undesired effects on the IDE, which will be fixed later. This PR is focused on restricting actual AST modifications.
- Moving nodes (and updating metadata in general) no longer causes reevaluation

https://user-images.githubusercontent.com/6566674/231616408-4f334bb7-1985-43ba-9953-4c0998338a9b.mp4

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/src/controller/graph/executed.rs
Comment (farmaazon) [inline app/gui/src/controller/graph/executed.rs]: Please make else block instead of return here and everywhere else. It should not obscure the function body much, and we avoid early return

**Review by farmaazon — APPROVED** [review]
Please test it before merge! The checkboxes are not checked.

File: app/gui/src/controller/graph/executed.rs
Comment (vitvakatu) [reply app/gui/src/controller/graph/executed.rs]: Done

**Comment by MichaelMauderer** [comment]
### QA report 🔴

When opening a project with this PR the state gets stuck at "Compiling Standard Library". This is not the case on `develop`. 


image

**Comment by vitvakatu** [comment]
@MichaelMauderer I can't reproduce it. I tried two different engines (3.30 and 4.13), and also tried both this branch and the current develop.

I triggered a similar-looking bug only once, on `develop` and engine `4.13`. I can't reproduce it anymore, though. Could you check if it is persistent for you, and if it happens with the specific project or not?

**Comment by MichaelMauderer** [comment]
> anymore, though. Could you check if it is persistent for you, and if it happens with the specific project or not?

I just tried this on another machine with (a) a different project and (b) a new empty project. I'm running into the same issue there. Maybe we should get someone else to check this? Could it be platform-specific?

**Comment by vitvakatu** [comment]
I found the issue, and it looks like the Engine behavior causes it. I wonder if I triggered the precisely same problem @MichaelMauderer has. I found only one reliable way to reproduce the issue: creating a new project. Opening an existing project caused the issue only once in 20+ tries, and I don't have logs for it (because I forgot to open dev tools in advance then).

What I see in network logs when I create a new project:
1. We create execution context.
2. We `applyEdit` with `execute: true` (it corresponds to adding an import statement for `Standard.Visualization`)
3. A bit later, we `applyEdit` with `execute: false` (as we update the metadata only)
4. The Engine never replies with `executionComplete`.

Screenshot 2023-04-18 at 13 52 47


When I open the existing project:
1. There are no `applyEdits` from us.
2. The Engine eventually replies with `executionComplete,` and everything is okay.

Screenshot 2023-04-18 at 13 56 10

So it looks like the Engine stops execution if we pass `execute: false` in one of the `applyEdits`. I suspect this is invalid, but I might be wrong. @4e6, could you take a look at this? I can reproduce it locally with this branch and engine version `4.18`.

**Review by wdanilo — APPROVED** [review]

**Comment by vitvakatu** [comment]
Merge is blocked by https://github.com/enso-org/enso/issues/6346

**Comment by wdanilo** [comment]
@vitvakatu if its blocked, please move it to draft PR until its unblocked. This way we easily see which PRs are ready for review/qa/merge :)

**Comment by vitvakatu** [comment]
Merged with develop, I can't reproduce the issue anymore.

**Comment by Procrat** [comment]
QA: Looks good to me (including creating a new project)!

**Pull request merged**
